From 88dbfb93194f15d0a894be713b8f9cb2ea7098a8 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Thu, 9 Dec 2021 09:54:34 +0100
Subject: [PATCH] Link rt with name and not with absolute path

---
 code/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/code/CMakeLists.txt b/code/CMakeLists.txt
index e45bf8a2aa..42743fe672 100644
--- a/code/CMakeLists.txt
+++ b/code/CMakeLists.txt
@@ -1300,7 +1300,7 @@ ENDIF()
 
 # Add RT-extension library for glTF importer with Open3DGC-compression.
 IF (RT_FOUND AND ASSIMP_IMPORTER_GLTF_USE_OPEN3DGC)
-  TARGET_LINK_LIBRARIES(assimp ${RT_LIBRARY})
+  TARGET_LINK_LIBRARIES(assimp rt)
 ENDIF ()
 
 
